cli/src/cli_util.rs

```diff
@@ -1053,6 +1053,12 @@ impl WorkspaceCommandHelper {
         let expression = revset::parse(
             revision_str,
             &self.revset_aliases_map,
+            // UserSettings::user_name() returns a fake user name if user.name is not configured
+            self.settings
+                .config()
+                .get_string("user.name")
+                .ok()
+                .as_deref(),
```

Comment:
If a user name/email isn't configured, I kinda think it's fair to treat any commits with that fake user as "mine", since if any commits were made without configuring the user, that's what  the commits would have as their author.

e.g. If I install JJ brand new, make a repo, and commit, I would expect `jj log -r 'mine()'` to show that commit.

Reply:
Does that point to a hole in the logic? Should jj not allow a commit without a defined user?

Reply:
Indeed, it does seem to make Git (or at least libgit2) unhappy:

```
❯ jj --config-toml 'user.email=""' --config-toml 'user.name=""' new root
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -1, klass: 3, message: "failed to parse signature - Signature cannot have an empty name or email" }', lib/src/git_backend.rs:266:46
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I guess I was thinking more along the lines of prompting for user info when it isn't defined.

Reply:
With jj's model of having an auto-amended commit for the working copy, that gets really annoying :)

Reply:
I think the placeholder name is fine, and I'll amend the commit to treat it as a real user email.

Possible conversation for later: I'm a bit uncomfortable using the placeholder names as magic phrases during pushing. What if they change or my repository is colocated and I later use `git push` directly?

Reply:
> Possible conversation for later: I'm a bit uncomfortable using the placeholder names as magic phrases during pushing. What if they change or my repository is colocated and I later use `git push` directly?

We already prevent that:
https://github.com/martinvonz/jj/blob/2d81cf91560ce891d3150e90227442d6dbdf449e/cli/src/commands/git.rs#L908-L914

Reply:
Those lines are what I was referring to. If the return value of `UserSettings::user_[id]_placeholder()` ever changes, then those checks will fail to recognize the old placeholder.

Reply:
Ah, I see. Yeah, I guess we'll just have to remember to also check the old names there then. Feel free to send a PR replacing those functions calls by string literals and a comment explaining it. Or maybe add a comment in a test case saying that it shouldn't be updated (convert it to a non-insta test).

Reply:
Actually, it's probably better to move those strings into the UI layer as just rendering artifacts. I'm thinking we should go back to using empty strings in [the model](https://github.com/martinvonz/jj/blob/2d81cf91560ce891d3150e90227442d6dbdf449e/lib/src/backend.rs#L140-L141) and then deal with Git not accepting empty strings in the Git backend, perhaps by setting some other placeholder there. Since that placeholder only shows up when using `git` (and other Git-adjacent tools) to interact with the repo, I think it's less important that it's user-friendly. Using e.g. "JJ_EMPTY_STRING_PLACEHOLDER" or something seems good enough to me. What do you think?

Reply:
Oh, one reason I meant to mention for why it's better to move it to the UI layer is i18n.

Reply:
> If a user name/email isn't configured, I kinda think it's fair to treat any commits with that fake user as "mine", since if any commits were made without configuring the user, that's what the commits would have as their author.
> 
> e.g. If I install JJ brand new, make a repo, and commit, I would expect `jj log -r 'mine()'` to show that commit.

Done. It now uses whatever `UserSettings::user_email()` returns.